field/binary.go

```diff
@@ -65,6 +65,10 @@ func (f *Binary) Pack() ([]byte, error) {
 		return nil, fmt.Errorf("failed to encode content: %w", err)
 	}
 
+	if len(packed) == 0 {
```

Comment:
I think that makes sense! Not sure why we didn't add it before :)
Also, it would be great to see a test for checks in this and other files. Just test that for a zero value it returns an empty slice.

Reply:
No problem - added a small test now